src/btree/bt_discard.c

```diff
@@ -240,6 +240,28 @@ __free_page_modify(WT_SESSION_IMPL *session, WT_PAGE *page)
     __wt_free(session, page->modify);
 }
 
+/*
+ * __ref_addr_safe_free --
+ *     Any thread that is reviewing the address in a WT_REF, must also be holding a split generation
+ *     to ensure that the page index they are using remains valid. Use that same split generation to
```

Comment:
This does seem contradictory to what is in the code, similar to what Alex G noted below. If we were using the same generation we should call "__wt_session_gen" should we improve the comment or am I misunderstanding something?

Reply:
The same split generation means `WT_GEN_SPLIT` using here and not the same generation number. The use of `__wt_gen` is correct. I will check if I can improve the comment to avoid this confusion.

Reply:
Perhaps: "Utilize the same generation type to safely free the address once all users of it have left the generation."

Reply:
Thanks. Fixed as suggested.